mlflow/gateway/providers/togetherai.py

```diff
@@ -0,0 +1,470 @@
+        # streaming and standard chat contain the same parameters
+        return TogetherAIAdapter.chat_to_model(payload, config)
+
+    @classmethod
+    def embeddings_to_model(cls, payload, config):
+        #Example request (https://docs.together.ai/reference/embeddings):
+        #curl --request POST \
+        #   --url https://api.together.xyz/v1/embeddings \
+        #   --header 'accept: application/json' \
+        #   --header 'content-type: application/json' \
+        #   --data '
+        #   {
+        #     "model": "togethercomputer/m2-bert-80M-8k-retrieval",
+        #     "input": "Our solar system orbits the Milky Way galaxy at about 515,000 mph"
+        #   }
+
+        # This is just to keep the interface consistent the adapter
+        # class is not needed here as the togetherai request similar
+        # to the openAI one.
+
+        return payload
+
+class TogetherAIProvider(BaseProvider):
+    NAME = "TogetherAI"
+
+    def __init__(self, config: RouteConfig) -> None:
+        super().__init__(config)
+        if config.model.config is None or not isinstance(config.model.config, TogetherAIConfig):
+            # Should be unreachable
+            raise MlflowException.invalid_parameter_value(
+                "Invalid config type {config.model.config}"
+            )
+        self.togetherai_config:  TogetherAIConfig = config.model.config
+
+    @property
+    def base_url(self):
+        #togetherai seems to support only this url
+        return "https://api.together.xyz/v1"
+
+    @property
+    def auth_headers(self):
+        return {"Authorization": f"Bearer {self.togetherai_config.togetherai_api_key}"}
+
+    async def _request(self, path: str, payload: Dict[str, Any]) -> Dict[str, Any]:
+        return await send_request(
+            headers=self.auth_headers,
+            base_url=self.base_url,
+            path=path,
+            payload=payload,
+        )
+
+    async def _stream_request(
+        self,
+        path: str,
+        payload: Dict[str, Any]
+    ) -> AsyncGenerator[bytes, None]:
+        return send_stream_request(
+            headers=self.auth_headers,
+            base_url=self.base_url,
+            path=path,
+            payload=payload,
+        )
+
+
+    async def embeddings(self, payload: embeddings.RequestPayload) -> embeddings.ResponsePayload:
+        from fastapi.encoders import jsonable_encoder
+
+        payload = jsonable_encoder(payload, exclude_none=True)
+
+        resp = await self._request(
+            path="embeddings",
+            payload={
+                "model": self.config.model.name,
+                **TogetherAIAdapter.embeddings_to_model(payload, self.config)
+            }
+        )
+
+        return TogetherAIAdapter.model_to_embeddings(resp, self.config)
+
+    # WARNING CHANGING THE ORDER OF METHODS HERE FOR SOME REASON CAUSES AN ERROR
+    # completions MODULE IS INTERPERTED AS THE completions function
```

Comment:
Can we import `completions` module with different name like `from mlflow.gateway.schemas import completions as completions_schema`? We shouldn't rely on the order of method declaration.

Reply:
True! Will do!